### PR TITLE
Add `custom_dummy_tokens` to `PretrainedTransformerTokenizer`

### PR DESCRIPTION
**Pull request by tarohi24**
This commit aims to support #5597

Changes proposed in this pull request:

- Add `custom_dummy_tokens` argument to `PretrainedTransformerTokenizer`. Without that argument, initializing `PretrainedTransformerTokenizer` of some non-English models fail.

## Before submitting

- [x] I've read and followed all steps in the [Making a pull request](https://github.com/allenai/allennlp/blob/main/CONTRIBUTING.md#making-a-pull-request)
    section of the `CONTRIBUTING` docs.
- [x] I've updated or added any relevant docstrings following the syntax described in the
    [Writing docstrings](https://github.com/allenai/allennlp/blob/main/CONTRIBUTING.md#writing-docstrings) section of the `CONTRIBUTING` docs.
- [ ] If this PR fixes a bug, I've added a test that will fail without my fix.
- [x] If this PR adds a new feature, I've added tests that sufficiently cover my new functionality.

## After submitting

- [ ] All GitHub Actions jobs for my pull request have passed.
- [ ] **`codecov/patch`** reports high test coverage (at least 90%).
    You can find this under the "Actions" tab of the pull request once the other checks have finished.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## Unreleased
 
+- Added `custom_dummy_tokens` argument to `TestPretrainedTransformerTokenizer`.
```
Comment (epwalsh) [inline CHANGELOG.md:10]:
```suggestion
### Added

- Added `custom_dummy_tokens` argument to `TestPretrainedTransformerTokenizer`.
```

File: allennlp/data/tokenizers/pretrained_transformer_tokenizer.py [hunk 2]
```diff
@@ -53,6 +55,7 @@ def __init__(
         add_special_tokens: bool = True,
         max_length: Optional[int] = None,
         tokenizer_kwargs: Optional[Dict[str, Any]] = None,
+        custom_dummy_tokens: Optional[Tuple[str, str]] = None,
```
Comment (epwalsh) [inline allennlp/data/tokenizers/pretrained_transformer_tokenizer.py:58]: How about calling this `verification_tokens` instead?
```suggestion
        verification_tokens: Optional[Tuple[str, str]] = None,
```

**Review by epwalsh — CHANGES_REQUESTED** [review]
Thanks @tarohi24! This looks good, I just have two small suggestions

**Comment by tarohi24** [comment]
Thanks @epwalsh for reviewing! Both suggestions look good to me. I committed them.

**Comment by tarohi24** [comment]
Thanks, @epwalsh for reviewing! The suggestions look good to me. I made commits to reflect these suggestions.

**Review by epwalsh — APPROVED** [review]
LGTM! Thanks @tarohi24!

**Comment by epwalsh** [comment]
Looks like we'll need to resolve https://github.com/allenai/allennlp/pull/5613 before merging this.

**Pull request merged**